Snowflake.Data.Tests/IntegrationTests/SFPutGetTest.cs

```diff
@@ -474,29 +517,40 @@ private static bool IsCompressedByTheDriver()
 
                 // Upload file
                 command.CommandText = putQuery;
-                var reader = command.ExecuteReader();
-                Assert.IsTrue(reader.Read());
-
-                // Check file status
-                Assert.AreEqual(expectedStatus.ToString(),
-                    reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ResultStatus));
-                // Check source and destination compression type
-                if (t_autoCompress)
+                try
                 {
-                    Assert.AreEqual(t_sourceCompressionType,
-                        reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.SourceCompressionType));
-                    Assert.AreEqual(t_destCompressionType,
-                        reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.DestinationCompressionType));
+                    var reader = command.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+                    // Checking query id when reader succeeded
+                    queryId = ((SnowflakeDbDataReader)reader).GetQueryId();
+                    // Checking if query Id is provided on the command level as well
+                    Assert.AreEqual(queryId, ((SnowflakeDbCommand)command).GetQueryId());
+                    // Check file status
+                    Assert.AreEqual(expectedStatus.ToString(),
```

Comment:
now all these assert don't work, because exception is caught and not rethrown

Reply:
if you refer to the exception from reader.Read how not rethrowing exception would affect asserts which are skipped in case of this exception?
Those asserts take place if there is a successful scenario.
Also all the tests pass right now. If there was an issue we would see some failures

Reply:
command.ExecuteReader() does not throw any exception, but reader.Read() return false.
In that case previously the test would fail. Currently the test will pass.
I would prefer to add try/catch only around execute and return the function immediately and not putting these asserts into try/catch block, because it makes some scenarios untested.